### PR TITLE
azure - updated functions structure & updated azure-mgmt-resource

### PR DESCRIPTION
**Pull request by logachev**
- Update version for azure.mgmt.resource package
- Update Azure Functions package structure

### CODE REVIEW

File: tools/c7n_azure/tests_azure/test_lock_action.py [hunk 1]
```diff
@@ -148,21 +151,23 @@ def test_lock_action_resource_group(self):
                 {
                     'type': 'lock',
                     'lock-type': 'CanNotDelete',
-                    'lock-name': 'testLock',
+                    'lock-name': 'testLock2',
```
Comment (aluong) [inline tools/c7n_azure/tests_azure/test_lock_action.py]: did we need to rename the lock?

File: tools/c7n_azure/tests_azure/tests_resources/test_subscription.py [hunk 2]
```diff
@@ -82,6 +84,12 @@ def test_add_policy(self, definition_patch):
         resources = p.run()
         self.assertEqual(len(resources), 1)
 
+        # Create Policy Assignment API might return before it was actually created
+        # Insert a delay to keep test active during nightly runs until this is fixed:
+        # https://github.com/Azure/azure-rest-api-specs/issues/7691
+        if not self.is_playback():
```
Comment (aluong) [inline tools/c7n_azure/tests_azure/tests_resources/test_subscription.py]: I think a better approach is to attempt to fetch and retry w/ a 3 second sleep than to just arbitrarily sleep.

**Comment by kapilt** [comment]
this appears to be causing trunk ci failures.

**Comment by kapilt** [comment]
its also unclear what the reasoning is for the archive layout changes

**Pull request merged**